kafka-connect-hive/src/main/scala/com/landoop/streamreactor/connect/hive/kerberos/KerberosLogin.scala

```diff
@@ -0,0 +1,107 @@
+package com.landoop.streamreactor.connect.hive.kerberos
+
+import java.io.IOException
+import java.net.InetAddress
+import java.security.PrivilegedAction
+
+import scala.concurrent.duration._
+import com.landoop.streamreactor.connect.hive.AsyncFunctionLoop
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.security.auth.login.LoginContext
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.security.SecurityUtil
+import org.apache.hadoop.security.UserGroupInformation
+
+import scala.concurrent.duration.Duration
+
+sealed trait KerberosLogin extends StrictLogging with AutoCloseable {
+  def run[T](thunk: => T):T
+}
+
+case class UserPasswordLogin(ugi: UserGroupInformation, interval: Duration, lc: LoginContext) extends KerberosLogin {
+  private val asyncTicketRenewal = new AsyncFunctionLoop(interval, "Kerberos")(renewKerberosTicket())
```

Comment:
I think you should call start on this unless it's somewhere else

Reply:
This executes with every put. So rather than run maybe `call` /`execute`

Reply:
There's a start method on the renewal but you never invoke it.

Reply:
true. done it